packages/cli/src/commands/extensions/install.ts

```diff
@@ -0,0 +1,167 @@
+  if (source.startsWith('http://') || source.startsWith('https://')) {
+    try {
+      const url = new URL(source);
+      const allowedHosts = [
+        'github.com',
+        'gitlab.com', 
+        'bitbucket.org',
+        'git.sr.ht',
+        'codeberg.org'
+      ];
+      return allowedHosts.includes(url.hostname);
+    } catch {
+      return false; // Invalid URL
+    }
+  }
+
+  return false;
+}
+
+export async function installExtension(argv: {
+  source: string;
+  project?: boolean;
+  user?: boolean;
+}) {
+  const { source } = argv;
+  const scope = getScope(argv);
+
+  let sourcePath: string;
+  let tempDir: string | undefined;
+  let isLocalPath = false;
+
+  try {
+    // Check if the source is a local path
+    if (fs.existsSync(source)) {
+      sourcePath = path.resolve(source);
+      isLocalPath = true;
+      console.log(`Installing local extension from ${sourcePath}...`);
+    } else if (isGitUrl(source)) {
+      // It's a git repository
+      tempDir = fs.mkdtempSync(path.join(os.tmpdir(), 'gemini-cli-ext-'));
+      console.log(`Cloning from ${source} into ${tempDir}...`);
+      execFileSync('git', ['clone', source, '.'], {
+        cwd: tempDir,
+        stdio: 'inherit',
+      });
+      sourcePath = tempDir;
+    } else {
+      throw new Error(
+        `Invalid source: "${source}". Must be a local path or git URL.`,
+      );
+    }
+
+    const extensionName = await getExtensionName(sourcePath);
+
+    const extensionsDir =
+      scope === 'user'
+        ? path.join(os.homedir(), EXTENSIONS_DIRECTORY_NAME)
+        : path.join(process.cwd(), EXTENSIONS_DIRECTORY_NAME);
+
+    if (!fs.existsSync(extensionsDir)) {
+      fs.mkdirSync(extensionsDir, { recursive: true });
+    }
+
+    const targetPath = path.join(extensionsDir, extensionName);
+
+    if (fs.existsSync(targetPath)) {
+      console.error(
+        `Extension "${extensionName}" is already configured within the ${scope} scope.`,
+      );
+      return;
+    }
+
+    fs.cpSync(sourcePath, targetPath, { recursive: true });
+
+    const settingsManager = new SettingsManager(scope);
+    const metadata: ExtensionMetadata = {
+      name: extensionName,
+      source: isLocalPath ? targetPath : source,
+      installDate: new Date().toISOString(),
+      lastUpdated: new Date().toISOString(),
+      active: true,
+      scope,
+    };
+
+    await settingsManager.addExtension(metadata);
+
+    console.log(`Extension "${extensionName}" installed in ${scope} scope.`);
+  } finally {
+    if (tempDir) {
+      fs.rmSync(tempDir, { recursive: true, force: true });
+    }
+  }
+}
+
+export const installCommand: CommandModule = {
+  command: 'install <source>',
+  describe: 'Install an extension from a Git URL or local path',
+  builder: (yargs) =>
+    yargs
+      .positional('source', {
+        describe: 'Git URL or local path of the extension to install',
+        type: 'string',
+        demandOption: true,
+      })
+      .option('project', {
```

Comment:
instead of two scopes arg options we should just take in a single argument `scope` that has values project or user etc.